### PR TITLE
moved spruce functionality into library form

### PR DESCRIPTION
**Pull request by thomasmmitchell**
Should now be accessible from `github.com/geofffranks/spruce`

### CODE REVIEW

**Comment by longnguyen11288** [comment]
does this break pipeline for any reason?

**Comment by thomasmmitchell** [comment]
It shouldn't - the pipeline references everything I can see by `./...`, so the changing of paths within the project folder _shouldn't_ cause an issue.

**Comment by kstrini** [comment]
It broke the homebrew install

**Comment by geofffranks** [comment]
I will take a look at this.

**Comment by geofffranks** [comment]
This branch didn't, the issue is in the concourse pipeline. Pipeline + homebrew should be fixed later today.

File: cmd/main.go [hunk 1]
```diff
@@ -39,36 +43,8 @@ var usage = func() {
...
-		content := fmt.Sprintf(format, args...)
-		lines := strings.Split(content, "\n")
-		for i, line := range lines {
-			lines[i] = "-----> " + line
-		}
-		content = strings.Join(lines, "\n")
-		printfStdErr("%s\n", content)
-	}
-}
-
```
Comment (geofffranks) [inline cmd/main.go]: Where did these tests go?

**Pull request merged**